sdk/core/azure-core/src/samples/java/com/azure/core/util/AsyncCloseableJavaDocCodeSnippet.java

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.util;
+
+import reactor.core.publisher.Flux;
+import reactor.core.publisher.Mono;
+import reactor.core.publisher.Sinks;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.stream.IntStream;
+
+/**
+ * Code snippets for {@link AsyncCloseable}.
+ */
+public class AsyncCloseableJavaDocCodeSnippet {
+    public void asyncResource() throws IOException {
+        // BEGIN: com.azure.core.util.AsyncCloseable.closeAsync
```

Comment:
Would it be better for this sample to use the same sample in the gist with a `usingWhen`?

Reply:
Ahh, I had used that in my sample because it was in a main method, so I wanted it to end. I expect typically people will hold a reference to the actual class and programmatically call closeAsync()